src/agent/debuglet.ts

```diff
@@ -164,6 +164,9 @@ export class Debuglet extends EventEmitter {
   private project: string|null;
   private controller: Controller;
   private completedBreakpointMap: {[key: string]: boolean};
+  private warningListener: ((warning: NodeJS.ErrnoException) => void)&{
+    fromGcpDebugAgent: boolean;
```

Comment:
`fromGcpDebugAgent` is only for unit tests, right? Would it make more sense to make it an optional field then?

I'm not very happy with adding a debugging-only property here but I don't have any other good ideas for doing that.

Reply:
I can make it optional.  I also don't like this solution, but don't know of anything better to assist in tests.  I think just counting the number of listeners is too fragile.

Reply:
I've uploaded another solution.